packages/at_secondary_server/test/sync_unit_test.dart

```diff
@@ -0,0 +1,724 @@
+            '@alice:phone@alice',
+            AtData()
+              ..data = '345'
+              ..metaData = (AtMetaData()..ttl = 10000));
+        // Assert the metadata
+        AtData? atDataAfterUpdate = await secondaryPersistenceStore!
+            .getSecondaryKeyStore()!
+            .get('@alice:phone@alice');
+        expect(atDataAfterUpdate!.data, '345');
+        expect(atDataAfterUpdate.metaData!.ttl, 10000);
+        // At the moment, the version and updatedAt have incorrect values which is known bug.
+        // Hence not adding the validations.
+        // Assert the commit entry
+        Iterator itr = atCommitLog!.getEntries(-1);
+        while (itr.moveNext()) {
+          expect(itr.current.value.operation, CommitOp.UPDATE_ALL);
+          expect(itr.current.value.commitId, 1);
+        }
+      });
+
+      test(
+          'test to verify when the commit entry is update_meta, the key metadata alone is updated',
+          () async {
+        ///Precondition
+        /// 1. The key already exists in the cloud secondary keystore
+        /// 2. The version of the key is 0
+        ///
+        /// Assertions:
+        /// 1. A new key is created in the hive keystore
+        /// 2. The commit log should have a new entry
+        /// 3. The version of the key is set to 1
+        /// 4. The "createdAt" is less than now()
+        /// 5. The "updatedAt" is populated and is less than now()
+        /// 6. The "createdBy" is assigned to currentAtSign
+        /// 7. update_meta commit entry is received where commit entry contains change in metadata fields
+        // Inserting a new key into keystore
+        await secondaryPersistenceStore!
+            .getSecondaryKeyStore()
+            ?.put('@alice:phone@alice', AtData()..data = '123');
+        // Updating the existing key
+        await secondaryPersistenceStore!
+            .getSecondaryKeyStore()
+            ?.putMeta('@alice:phone@alice', AtMetaData()..ttl = 10000);
+        // verify the metadata
+        AtData? atData = await secondaryPersistenceStore!
+            .getSecondaryKeyStore()!
+            .get('@alice:phone@alice');
+        expect(atData!.metaData!.ttl, 10000);
+        // Verify commit entry
+        CommitEntry? commitEntryList =
+            atCommitLog!.getLatestCommitEntry('@alice:phone@alice');
+        expect(commitEntryList!.operation, CommitOp.UPDATE_META);
+        expect(commitEntryList.commitId, 1);
+      });
+
+      test(
+          'test to verify when the commit entry is delete, the key is deleted from hive keystore and entry added to commit log',
+          () async {
+        ///Precondition
+        /// 1. The serverCommitId is at 5
+        /// 2. The key already exists in cloud secondary
+        /// 3. delete commit entry is received
+        ///
+        /// Assertions:
+        /// 1. The should be deleted from the keystore
+        /// 2. The commit log should be updated
+        await secondaryPersistenceStore!
```

Comment:
Where is the key "@alice:phone@alice" inserted before the delete? Should the commit ID and version be 1?

"Please check the code and comments and see that they are in sync"

Reply:
This change is implemented